_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,316 @@
+is limited to the namespace *istio-system1*. A completed manifest would contain many
+more `apiGroups` under the `Role` providing resource access to the tenant administrator.
+```yaml
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: istio-system1 
+  name: ns-access-for-sales-admin-istio-system1
+rules:
+- apiGroups: [""] # "" indicates the core API group
+  resources: ["*"]
+  verbs: ["*"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: access-all-istio-system1
+  namespace: istio-system1
+subjects:
+- kind: User
+  name: sales-admin
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: ns-access-for-sales-admin-istio-system1
+  apiGroup: rbac.authorization.k8s.io
+
+```
+
+### Watching specific namespaces for service discovery
+In addition to creating RBAC rules limiting the tenant administrator access to a specific Istio
+control plane, the Istio manifest must be updated to specify the application namespace that
+Pilot should watch for creation of its xDS cache. This is done by starting the Pilot component
+with the additional command line arguments `--appNamespace, ns-1`.  Where *ns-1* is the
+namespace that the tenant’s application will be deployed in. An example snippet from the
+istio-system1.yaml file is included below.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-pilot
+  namespace: istio-system1
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        istio: pilot
+    spec:
+      serviceAccountName: istio-pilot-service-account
+      containers:
+      - name: discovery
+        image: docker.io/<user ID>/pilot:<tag>
+        imagePullPolicy: IfNotPresent
+        args: ["discovery", "-v", "2", "--admission-service", "istio-pilot", "--appNamespace", "ns-1"]
+        ports:
+        - containerPort: 8080
+        - containerPort: 443
+
+```
+
+### Deploying the tenant application in a namespace
+Now that the cluster administrator has created the tenant's namespace (ex. *istio-system1*) and
+Pilot's service discovery has been configured to watch for a specific application
+namespace (ex. *ns-1*), create the application manifests to deploy in that tenant's specific
+namespace. For example:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns-1
+```
+And add the namespace reference to each resource type included in the applications manifest
+file.  For example:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: details
+  labels:
+    app: details
+  namespace: ns-1
+```
+Although not shown, the application namespaces will also have RBAC settings limiting access
+to certain resources. These RBAC settings could be set by the cluster administrator and/or
+the tenant administrator.
+
+### Using Istioctl commands in a multitenant environment
```

Comment:
Istioctl commands in -> `istioctl` in

Reply:
Done. (Although it looks like you emphasized 'istioctl', not sure if you wanted that emphasis in the headline or you were just trying to get my attention to remove the capitalization.)